numpy/core/tests/test_multiarray.py

```diff
@@ -4728,6 +4728,26 @@ def test_vecself(self):
         p = np.inner(a, a)
         assert_almost_equal(p, 0, decimal=14)
 
+    def test_inner_product_with_various_contiguities(self):
+        # github issue 6532
+        for dtypes, dt_specific_assert_equal in (
+                (np.typecodes['AllInteger'], assert_equal),
```

Comment:
Too complicated I think. `assert_equal` should be fine for everything as all the numbers are exactly representable as floats. The boolean tests should probably be separate as all the arrays elements are true in that case. So the loop can be
```
for dt in  np.typecodes['AllInteger'] + np.typecodes['AllFloat']:
```